src/validation/input-validator.ts

```diff
@@ -140,26 +142,62 @@ export class InputValidator {
     }
 
     if (['array', 'object'].includes(type)) {
-      if (
-        type === 'array' &&
-        (!Array.isArray(normalized[key]) || (normalized[key] as unknown[]).length === 0)
-      ) {
-        this.throwInvalid(`${key} parameter must be a non-empty array`)
+      if (type === 'array') {
+        this.validateArray(normalized[key] as unknown[], key)
       }
-      if (
-        type === 'object' &&
-        normalized[key] &&
-        (isNotObject(normalized[key] as Record<string, unknown>) ||
-          Object.keys(normalized[key] as Record<string, unknown>).length === 0)
-      ) {
-        this.throwInvalid(`${key} parameter must be an object with at least one property`)
+      if (type === 'object') {
+        this.validateObject(normalized[key] as Record<string, unknown>, key)
       }
       return
     } else if (typeof normalized[key] !== type) {
       this.throwInvalid(`${key} parameter must be of type ${type}`)
     }
   }
 
+  private validateArray(data: unknown[], key: string) {
+    if (!Array.isArray(data) || data.length === 0) {
+      this.throwInvalid(`${key} parameter must be a non-empty array`)
+    }
+
+    if (data.length > MAX_ARRAY_LENGTH) {
+      this.throwInvalid(
+        `${key} parameter must be an array of length less than ${MAX_ARRAY_LENGTH}. Received ${data.length} items`,
+      )
+    }
+
+    data.forEach((item, index) => {
+      if (!['string', 'number', 'boolean'].includes(typeof item)) {
+        this.throwInvalid(
+          `${key} parameter must be an array of strings|numbers|booleans, received '${typeof item}' at index ${index}`,
+        )
+      }
+    })
+  }
```

Comment:
I could see these validations causing issues for PoR adapters on v3. Depending on the pool of addresses for a client, we could have an array with thousands of addresses. And with the current format, it would be an array of objects. Maybe we could make max items configurable and also allow objects? Something to maybe discuss with Alejo. An example is something like below.
```
{
    "data": {
        "addresses": [
            ...
            {
                "address": "0x0000"
            },
            ...
        ]
    }
}
```

Reply:
@alejoberardino any thoughts on this ?